index.js

```diff
@@ -1 +1,438 @@
+ * and comparing the resulting array.
+ *
+ * Needed because IE11 doesn't support Set#entries or Set#@@iterator
+ *
+ * @param {Set} a
+ * @param {Set} b
+ * @return {Boolean} result
+ */
+
+function setEqual(leftHandOperand, rightHandOperand, options) {
+  var leftHandItems = [];
+  var rightHandItems = [];
+  leftHandOperand.forEach(function gatherSetEntries(entry) {
+    leftHandItems.push(entry);
+  });
+  rightHandOperand.forEach(function gatherSetEntries(entry) {
+    rightHandItems.push(entry);
+  });
+  return iterableEqual(leftHandItems.sort(), rightHandItems.sort(), options);
+}
+
+/*!
+ * Compare two maps by forEaching over them,
+ * and comparing the resulting key/value pairs.
+ *
+ * Speed Optimization:
+ * Pre:
+ *   map                         x 320,939 ops/sec ±1.45% (81 runs sampled)
+ *   map (complex)               x 149,099 ops/sec ±1.55% (76 runs sampled)
+ *   map (differing)             x 251,749 ops/sec ±7.59% (73 runs sampled)
+ * Post:
+ *   map                         x 511,685 ops/sec ±1.70% (82 runs sampled)
+ *   map (complex)               x 247,723 ops/sec ±1.42% (81 runs sampled)
+ *   map (differing)             x 450,797 ops/sec ±1.74% (78 runs sampled)
+ * @param {Set} a
+ * @param {Set} b
+ * @return {Boolean} result
+ */
+
+function mapEqual(leftHandOperand, rightHandOperand, options) {
+  if (leftHandOperand.size !== rightHandOperand.size) {
+    return false;
+  }
+  var result = true;
+  leftHandOperand.forEach(function traverseLHMap(lhoKey, lhoValue) {
+    var found = false;
+    rightHandOperand.forEach(function traverseRHMap(rhoKey, rhoValue) {
+      if (deepEqual(lhoKey, rhoKey, options) && deepEqual(lhoValue, rhoValue, options)) {
+        found = true;
+      }
+    });
+    if (!found) {
+      result = false;
+    }
+  });
+  return result;
+}
+
+/*!
+ * Simple equality for flat iterable objects
+ * such as Arrays, TypedArrays or Node.js buffers.
+ *
+ * @param {Iterable} leftHandOperand
+ * @param {Iterable} rightHandOperand
+ * @return {Boolean} result
+ */
+
+function iterableEqual(leftHandOperand, rightHandOperand, options) {
+  var length = leftHandOperand.length;
+  if (length !== rightHandOperand.length) {
+    return false;
+  }
+  for (var i = 0; i < length; i += 1) {
+    if (deepEqual(leftHandOperand[i], rightHandOperand[i], options) === false) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/*!
+ * Simple equality for generator objects
+ * such as those returned by generator functions.
+ *
+ * @param {Iterable} leftHandOperand
+ * @param {Iterable} rightHandOperand
+ * @return {Boolean} result
+ */
+
+function generatorEqual(leftHandOperand, rightHandOperand, options) {
+  return iterableEqual(getGeneratorEntries(leftHandOperand), getGeneratorEntries(rightHandOperand), options);
+}
+
+/*!
+ *
+ *
+ *
+ */
+function getEnumerableKeys(target) {
+  try {
+    return Object.keys(target);
+  } catch (keysError) {
+    return [];
+  }
+}
```

Comment:
I think neither are needed.
Since all the primitives have been treated before and the only place this function gets called is when we're trying to compare `objects` we can just thrust `target` won't throw a `TypeError`.